lib/experimental/fonts/font-library/class-wp-font-collection.php

```diff
@@ -48,184 +47,158 @@ class WP_Font_Collection {
+
+		if ( empty( $args['font_families'] ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Font collection slug. */
+				sprintf( __( 'Font collection "%s" does not contain any font families.', 'gutenberg' ), $slug ),
+				'6.5.0'
+			);
+		}
 	}
 
 	/**
-	 * Checks if the font collection config is valid.
+	 * Loads the font collection data from a json file path or url.
 	 *
 	 * @since 6.5.0
 	 *
-	 * @param array $config Font collection config options. {
-	 *      @type string $slug        The font collection's unique slug.
-	 *      @type string $name        The font collection's name.
-	 *      @type string $description The font collection's description.
-	 *      @type string $src         The font collection's source.
-	 *      @type array  $font_families An array of font families in the font collection.
-	 *      @type array  $categories The font collection's categories.
-	 *  }
-	 * @return bool True if the font collection config is valid and false otherwise.
+	 * @param string $file_or_url File path or url to a json file containing the font collection data.
+	 * @return array|WP_Error An array containing the font collection data on success,
+	 *                        else an instance of WP_Error on failure.
 	 */
-	public static function is_config_valid( $config ) {
-		if ( empty( $config ) || ! is_array( $config ) ) {
-			_doing_it_wrong( __METHOD__, __( 'Font Collection config options are required as a non-empty array.', 'gutenberg' ), '6.5.0' );
-			return false;
-		}
+	public static function load_from_json( $file_or_url ) {
+		$url  = wp_http_validate_url( $file_or_url );
+		$file = file_exists( $file_or_url ) ? wp_normalize_path( realpath( $file_or_url ) ) : false;
 
-		$required_keys = array( 'slug', 'name' );
-		foreach ( $required_keys as $key ) {
-			if ( empty( $config[ $key ] ) ) {
-				_doing_it_wrong(
-					__METHOD__,
-					// translators: %s: Font collection config key.
-					sprintf( __( 'Font Collection config %s is required as a non-empty string.', 'gutenberg' ), $key ),
-					'6.5.0'
-				);
-				return false;
-			}
+		if ( ! $url && ! $file ) {
+			// translators: %s: File path or url to font collection json file.
+			return new WP_Error( 'font_collection_loading_failed', sprintf( __( 'Font collection JSON file "%s" is invalid or does not exist.', 'gutenberg' ), $file_or_url ) );
 		}
 
-		if (
-			( empty( $config['src'] ) && empty( $config['font_families'] ) ) ||
-			( ! empty( $config['src'] ) && ! empty( $config['font_families'] ) )
-		) {
-			_doing_it_wrong(
-				__METHOD__,
-				sprintf(
-					/* translators: %1$s: src, %2$s: font_families */
-					__( 'Font Collection config "%1$s" option OR "%2$s" option is required.', 'gutenberg' ),
-					'src',
-					'font_families'
-				),
-				'6.5.0'
-			);
-			return false;
+		$data = $url ? self::load_from_url( $url ) : self::load_from_file( $file );
+
+		if ( is_wp_error( $data ) ) {
+			return $data;
 		}
 
-		return true;
+		return $data;
```

Comment:
This seems to be returning $data anyways. What's the purpose of the if() ?

Reply:
Good point! Will remove the conditional.